modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -428,6 +447,128 @@ object ObservationService {
           _             <- ResultT(asterismService.setAsterism(pid, NonEmptyList.of(newOid), input.asterism))
         yield CloneIds(origOid, newOid)).value
         
+      // ***********************
+      // All about the validations
+      // ***********************
+      def cfpError(msg: String): ObservationValidation =
+        ObservationValidation.callForProposals(msg)
+
+      extension (ra: RightAscension)
+        def isInInterval(raStart: RightAscension, raEnd: RightAscension): Boolean =
+          if (raStart > raEnd) raStart <= ra || ra <= raEnd
+          else raStart <= ra && ra <= raEnd
+
+      extension (dec: Declination)
+        def isInInterval(decStart: Declination, decEnd: Declination): Boolean =
+          decStart <= dec && dec <= decEnd
+
+      extension (tsi: TimestampInterval)
+        def centerUnsafe: Timestamp =
+          if (tsi.isEmpty) tsi.start
+          else tsi.start.plusMicrosOption(tsi.boundedTimeSpan.toMicroseconds).get
```

Comment:
Is it unsafe because of the `get``?

Reply:
Yes. But in practice it should always be OK.

Reply:
Wouldn't the center be half of the interval?  I'm probably missing the obvious.

Reply:
Oh. 🤦 I forgot to divide by 2. I got too involved in figuring out where to get the data from...